### ISSUE TITLE
[Improvement] Negative z-index values

### ISSUE THREAD

**Issue by drank** [pojome/elementor#3677]
Please consider allowing negative z-index values for the widgets in Advanced. Thank you!

**Comment by bainternet** [comment]
@drank 

any specific use cases?

**Comment by drank** [comment]
@bainternet 

Please check the "Stacking Order" section in this article - https://webdesign.tutsplus.com/articles/what-you-may-not-know-about-the-z-index-property--webdesign-16892

It very well describes the use-cases when negative z-index would be needed.

In addition, negative z-index is a valid value as per the CSS specs (https://developer.mozilla.org/en-US/docs/Web/CSS/z-index)

**Comment by bainternet** [comment]
@drank 

If we allow negative then the element could disappear from the editor, that's our main problem with it.

as for use case, again, whats wrong with giving a higher value to the top element instead of negative to the bottom one?

**Comment by drank** [comment]
@bainternet 

OK, I understand about the editor limitations, but that's what we are trying to overcome, aren't we? :)

Please check the image from the article that I referred to above ([here is a link](https://cdn.tutsplus.com/webdesign/uploads/2013/11/stacking-order1.png)). If I want to put something below "Block Level Boxes", "Floated boxes" or "Inline Boxes" I should use a negative z-order.

Also, what if I want to put an element behind some multiple elements that are currently below that element. What is easier - to set one element to negative z-order or to change the z-order of multiple elements only to satisfy the Z-positioning of that element (in which case we transfer the z-order responsibility to the other elements)?

IMHO we (and Elementor) will definitely benefit from the ability to have negative z-order values.

**Comment by drank** [comment]
@bainternet, I do understand that now you are under pressure with 2.0 but I really hope that this topic does not get buried, along with my other request - https://github.com/pojome/elementor/issues/3679.

Please, let's work out a solution!

Thank you in advance!

**Comment by drank** [comment]
@bainternet , thanks for the code! :)

**Comment by zeinnicholas** [comment]
> @drank
> 
> Like I said we feel that with allowing negative z-index out of the box is not something we want, and as I don't see this happening anytime soon, And this is something You _must_ have ( only saying that because things can get messy real quick with negative z-index ), Then here is a snippet that will allow you to do that:

@bainternet I strongly disagree. I just found someone having issues with anchor widgets overlapping buttons (see: https://www.facebook.com/groups/Elementors/permalink/2475951502530621/) Probably they were using this snippet provided by your team in the last section here:
https://docs.elementor.com/article/62-menu-anchor

You should either provide a better solution or allow for simpler ways to set z-index to negative, which is what's being requested here. What will it be?

**Comment by wmodes** [comment]
Thank you for the useful patch code. Can I ask you to improve it a bit?

I have plenty of use cases for a negative z-index, particularly since setting a positive index on some elements creates a stacking context. Instead, we set others negative to pull them backward.

I understand and experienced the danger of setting a negative z-index. When I do, I can no longer edit the element because it falls behind the elementor canvas. Is it possible to allow a negative z-index in the display option, but *display* in in the editor with a default index? Or alternatively, set the elementor canvas (or whatever blocks editing) at -10000 or something?

**Comment by 75th** [comment]
Here's a use case:

<img width="349" alt="Screen Shot 2020-10-26 at 3 55 07 PM" src="https://user-images.githubusercontent.com/1731790/97227534-a1683000-17a3-11eb-972e-2c73a4810f76.png">

I need the image to be beneath the parent's border. This requires `z-index: -1`; a z-index of zero won't cut it.

**Comment by itumulak** [comment]
You should enable negative z-index. One use case is we have an image for the text, this needs to be under the text. Having `z-index: -1` on the circle marker makes the text on top.

![Web capture_21-1-2021_163446_projects invisionapp com](https://user-images.githubusercontent.com/2912946/105324671-a99b0480-5c06-11eb-839e-7c99b82a9792.jpeg)

**Comment by leecollings** [comment]
Can we please either enable this (the input of negative z-index values) or re-open this issue please for further discussion?

There's no reason not to do this, and this shouldn't be limited like this either. Negative z-index values are very much required at times (the alternative solution is to set a higher z-index value to EVERY other element apart from the one you want to edit... which is very counter-intuitive).

There are literally going to be thousands of people complaining about not being able to do this, and it's not going to cause any hurt at all, so there is quite literally no reason why this is like this?

Please get this done, because I now cannot do something because of this (and it's quite important!)
